azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -21,16 +21,17 @@
 
 from azurelinuxagent.common.exception import VmSettingsError
 from azurelinuxagent.common.protocol.extensions_goal_state import ExtensionsGoalState
+from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
 from azurelinuxagent.common.utils.textutil import format_exception
 
 
 class ExtensionsGoalStateFromVmSettings(ExtensionsGoalState):
     def __init__(self, etag, json_text):
         super(ExtensionsGoalStateFromVmSettings, self).__init__()
         self._id = etag
-        self._host_ga_plugin_version = None
-        self._schema_version = None
         self._text = json_text
+        self._host_ga_plugin_version = FlexibleVersion()
+        self._schema_version = FlexibleVersion()
```

Comment:
Can you please help me understand what FlexibleVersion() without any parameter passed returns and what will be the default value of _host_ga_plugin_version?
Had a quick read about FlexibleVersion(version.Version) and looks like it expects a version.

Reply:
self.version is intialized to empty and each property defaults to 0, as in

    def major(self):
        return self.version[0] if len(self.version) > 0 else 0